src/rhub/bare_metal/model/host.py

```diff
@@ -0,0 +1,198 @@
+)
+
+logger = logging.getLogger(__name__)
+
+
+class BareMetalHostStatus(str, enum.Enum):
+    AVAILABLE = "available"
+    ENROLLING = "enrolling"
+    FAILED_ENROLLING = "failed_enrolling"
+    MAINTENANCE = "maintenance"
+    RESERVED = "reserved"
+    UNAVAILABLE = "unavailable"
+
+
+class BareMetalHostVendor(str, enum.Enum):
+    GENERIC = "generic"
+    DRAC = "drac"
+    REDFISH = "redfish"
+
+
+class BareMetalHost(db.Model, ModelMixin, TimestampMixin):
+    __tablename__ = _BM_TABLE_NAME_HOST
+
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(128), unique=True, nullable=False)
+    mac = db.Column(db.String(20), nullable=False)
+
+    #: :type: :class:`BareMetalArch`
+    arch = db.Column(db.Enum(BareMetalArch), nullable=False)
+
+    # boot type - TODO: think of some bitwise operation
+    legacy_bios = db.Column(
+        db.Boolean, server_default=expression.true(), nullable=False
+    )
+    uefi = db.Column(db.Boolean, server_default=expression.true(), nullable=False)
+    secure_boot = db.Column(
+        db.Boolean, server_default=expression.true(), nullable=False
+    )
+
+    #: :type: :class:`BareMetalHostStatus`
+    status = db.Column(
+        db.Enum(BareMetalHostStatus),
+        server_default=BareMetalHostStatus.UNAVAILABLE.name,
+        nullable=False,
+    )
+
+    #: :type: :class:`BareMetalHostVendor`
+    type = db.Column(db.Enum(BareMetalHostVendor), nullable=False)
+
+    handler_id = db.Column(
+        db.Integer, db.ForeignKey(f"{_BM_TABLE_NAME_HANDLER}.id"), nullable=False
+    )
+    #: :type: :class:`BareMetalHandler`
+    handler = db.relationship("BareMetalHandler", back_populates="hosts")
+    handler_uuid = db.Column(postgresql.UUID)
+    handler_data = db.Column(db.JSON)
+
+    ipmi_address = db.Column(db.String(128), nullable=False)
+    ipmi_password = db.Column(db.String(128), nullable=False)
+    ipmi_port = db.Column(db.String(128), nullable=False)
+    ipmi_username = db.Column(db.String(128), nullable=False)
+
+    #: :type: :class:`BareMetalProvision`
+    deployments = db.relationship("BareMetalProvision", back_populates="host")
+
+    __mapper_args__ = {
+        "polymorphic_on": type,
+        "polymorphic_identity": BareMetalHostVendor.GENERIC,
+    }
+
+    @classmethod
+    def credential_columns(cls) -> [str]:
+        return ["ipmi_address", "ipmi_password", "ipmi_port", "ipmi_username"]
+
+    def get_credentials(self) -> dict[str, str]:
+        columns = (
+            self.credential_columns() + super(BareMetalHost, self).credential_columns()
+        )
+        data = {}
+        for column in columns:
+            value = getattr(self, column)
+            if value:
+                data[column] = value
+        return data
+
+    def is_available(self) -> bool:
+        return (
+            self.status is BareMetalHostStatus.AVAILABLE and self.handler.is_available()
+        )
+
+    def can_be_enrolled(self) -> bool:
```

Comment:
Confusing.   How can self.status be UNAVAILABLE and calls to is_available checks self.status is AVAILABLE.  Perhaps a few comments would help.

Reply:
I renamed the 'unavailable' to 'non-enrolled', hope it's clear.
TY for pointing that.